parser/parser.y

```diff
@@ -3325,6 +3326,11 @@ TableElement:
 	{
 		$$ = $1.(*coldef.TableConstraint)
 	}
+|	"CHECK" '(' Expression ')'
```

Comment:
no special reference in mysql document too, just list it in table create_definition.

So we can only refer http://dev.mysql.com/doc/refman/5.7/en/create-table.html here, but this seems no need.